synfig-core/src/synfig/node.h

```diff
@@ -286,6 +286,20 @@ class Node : public etl::rshared_object
 		return parent;
 	}
 
+	//! Return the first parent of a given type with an additional match condition
+	//! The CompareFunc should return true when match is satisfied
+	//! Example: node->find_first_parent_of_type<MyType>([](MyType::Handle v) -> bool { return v.prop == 1;});
+	template<typename T> etl::handle<T> find_first_parent_of_type(const sigc::slot<bool(etl::handle<T>)> &CompareFunc) const {
```

Comment:
`sigc::slot<bool(const etl::handle<T>&)` to make sure argument is read-only.

Reply:
Done